crypto/self_test_core.c

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright 2019 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <openssl/self_test.h>
+
+static OSSL_SELF_TEST_event_fn *ossl_self_test_cb = NULL;
+void OSSL_SELF_TEST_set_callback(OSSL_SELF_TEST_event_fn *cb)
```

Comment:
So, er, multiplexing?

Reply:
The original intention was that this mechanism was supposed to be an application callback that the user provides to receive information back when the self tests are running at startup for the FIPS module (I dont quite get why an OPENSSL_CTX is needed for this? - it runs from the provider so it knows its context). 
Do you also want to run self tests for the default provider (This would require quite a bit more work to figure out what needs to be passed to the default provider - i.e: I really dont want to do integrity checks on the default provider because it is all loaded via configuration entries).

Reply:
Hmm looking st this a bit more - if I use OSSL_CALLBACK I can kill 2 birds... since it already has an args param

Reply:
Note that the semantics of that void argument is that it's whatever the caller chose to pass along with the callback itself.

Reply:
What I'm thinking is that you could have more than one FIPS provider loaded at the same time into different OPENSSL_CTXs. You might want to set a callback on one of those OPENSSL_CTXs but not the other.